buildSrc/src/main/java/org/elasticsearch/gradle/test/RestIntegTestTask.java

```diff
@@ -0,0 +1,114 @@
+    private static final String TESTS_REST_CLUSTER = "tests.rest.cluster";
+    private static final String TESTS_CLUSTER = "tests.cluster";
+    private static final String TESTS_CLUSTER_NAME = "tests.clustername";
+
+    public RestIntegTestTask() {
+        Project project = getProject();
+        String name = getName();
+        super.dependsOn(project);
+        runner = project.getTasks().create(name + "Runner", RestTestRunnerTask.class);
+        NamedDomainObjectContainer<ElasticsearchCluster> testClusters = (NamedDomainObjectContainer<ElasticsearchCluster>) project
+            .getExtensions()
+            .getByName("testClusters");
+        ElasticsearchCluster cluster = testClusters.create(name);
+        runner.useCluster(cluster);
+        runner.include("**/*IT.class");
+        runner.systemProperty("tests.rest.load_packaged", Boolean.FALSE.toString());
+        if (System.getProperty(TESTS_REST_CLUSTER) == null) {
+            if (System.getProperty(TESTS_CLUSTER) != null || System.getProperty(TESTS_CLUSTER_NAME) != null) {
+                throw new IllegalArgumentException(
+                    String.format("%s, %s, and %s must all be null or non-null", TESTS_REST_CLUSTER, TESTS_CLUSTER, TESTS_CLUSTER_NAME)
+                );
+            }
+            SystemPropertyCommandLineArgumentProvider runnerNonInputProperties = (SystemPropertyCommandLineArgumentProvider) runner
+                .getExtensions()
+                .getByName("nonInputProperties");
+            runnerNonInputProperties.systemProperty(
+                TESTS_REST_CLUSTER,
+                (Supplier<String>) () -> String.join(",", cluster.getAllHttpSocketURI())
+            );
+            runnerNonInputProperties.systemProperty(
+                TESTS_CLUSTER,
+                (Supplier<String>) () -> String.join(",", cluster.getAllTransportPortURI())
+            );
+            runnerNonInputProperties.systemProperty(TESTS_CLUSTER_NAME, (Supplier<String>) cluster::getName);
+        } else {
+            if (System.getProperty(TESTS_CLUSTER) == null || System.getProperty(TESTS_CLUSTER_NAME) == null) {
+                throw new IllegalArgumentException(
+                    String.format("%s, %s, and %s must all be null or non-null", TESTS_REST_CLUSTER, TESTS_CLUSTER, TESTS_CLUSTER_NAME)
+                );
+            }
+            // an external cluster was specified and all responsibility for cluster configuration is taken by the user
+            runner.systemProperty(TESTS_REST_CLUSTER, System.getProperty(TESTS_REST_CLUSTER));
+            runner.systemProperty(TESTS_CLUSTER, System.getProperty(TESTS_CLUSTER));
+            runner.systemProperty(TESTS_CLUSTER_NAME, System.getProperty(TESTS_CLUSTER_NAME));
+        }
+        // this must run after all projects have been configured, so we know any project
+        // references can be accessed as a fully configured
+        project.getGradle().projectsEvaluated(x -> {
```

Comment:
Gross. Let's replace this with something like:

```
runner.onlyIf(t -> this.getEnabled())
```

Reply:
:( I tried it and ended up violating the testingConventions.
```
Expected at least one test class included in task :x-pack:plugin:autoscaling:integTestRunner, but found none.
```
I always wondered why these where needed: https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/autoscaling/build.gradle#L15  Fixing that is a bit more then i want to chew off for this PR, so put it back to how it was.

Reply:
😢 fair enough, we'll leave as-is